Update Helm docs to reflect 0.42.0 release (#12689)
hashicorp/consul: website/content/docs/k8s/helm.mdx

      `<global.name>-federation` (if setting `global.name`), otherwise
      `<helm-release-name>-consul-federation`.

    - `primaryDatacenter` ((#v-global-federation-primarydatacenter)) (`string: null`) - The name of the primary datacenter.

    - `primaryGateways` ((#v-global-federation-primarygateways)) (`array<string>: []`) - A list of addresses of the primary mesh gateways in the form `<ip>:<port>`.
      (e.g. ["1.1.1.1:443", "2.3.4.5:443"]

    - `k8sAuthMethodHost` ((#v-global-federation-k8sauthmethodhost)) (`string: null`) - If you are setting `global.federation.enabled` to true and are in a secondary datacenter,
      set `k8sAuthMethodHost` to the address of the Kubernetes API server of the secondary datacenter.
      This address must be reachable from the Consul servers in the primary datacenter.

>       This authmethod will be used to provision ACL tokens for Consul components and is different

Reviewer: ```suggestion
      This auth method will be used to provision ACL tokens for Consul components and is different
```

      from the one used by the Consul Service Mesh.

>       Please see the Kubernetes Auth Method documentation (https://consul.io/docs/acl/auth-methods/kubernetes).

Reviewer: ```suggestion
      Please see the [Kubernetes Auth Method documentation](https://consul.io/docs/acl/auth-methods/kubernetes) for additional information.
```

>       You could retrieve this value from your `kubeconfig` by running:

Reviewer: ```suggestion
      You can retrieve this value from your `kubeconfig` by running:
```
